iothub/device/src/IotHubClientOptions.cs

```diff
@@ -89,5 +89,22 @@ public string AdditionalUserAgentInfo
         }
 
         internal virtual ProductInfo UserAgentInfo { get; } = new();
+
+        internal IotHubClientOptions Clone()
+        {
+            IotHubClientTransportSettings transport = TransportSettings is IotHubClientMqttSettings settings
+                ? settings.Clone()
+                : ((IotHubClientAmqpSettings)TransportSettings).Clone();
```

Comment:
Can you put the clone method on the base class so they both get it and we don't have to cast?